### ISSUE TITLE
[Bug] Webview on iOS triggers too many navigation events

### ISSUE THREAD

**Issue by Mikilll94** [xamarin/Xamarin.Forms#10350]
### Description
WebView triggers too many navigation events on iOS.

I.e. when navigating to Microsoft website (https://www.microsoft.com/)

Here are the navigation events on Android:
Event:          e.Url
Navigating: https://www.microsoft.com/
Navigating: https://www.microsoft.com/pl-pl/
Navigated: https://www.microsoft.com/pl-pl/

And there the navigation events on iOS:
Event:          e.Url
Navigating: https://www.microsoft.com/
Navigating: https://www.microsoft.com/pl-pl/
Navigating: about:blank
Navigated: https://www.microsoft.com/pl-pl/
Navigating: https://mscom.demdex.net/dest5.html?d_nsid=0#https%3A%2F%2Fwww.microsoft.com%2Fpl-pl%2F
Navigating: https://login.live.com/me.srf?wa=wsignin1.0&wreply=https%3A%2F%2Fwww.microsoft.com&uaid=928a1a14-e6dc-489e-f3a6-17442576436a&partnerId=mshomepage
Navigating: https://www.microsoft.com/store/buy/cartcount

As you can see on iOS there is a lot more navigation events than on Android. And some of them are really strange, i.e. **Navigating: about:blank**.

### Steps to Reproduce

1. Run this project on Android
[WebViewTest.zip](https://github.com/xamarin/Xamarin.Forms/files/4488892/WebViewTest.zip)
2. Look at the console.
3. Run the same project on iOS
4. Look at the console again. As you can see on iOS there is a lot more navigation events than on Android. And some of them look like a trash.

### Expected Behavior
Navigation events on iOS should work correctly like on Android.

### Actual Behavior
Navigation events on iOS contains a lot of trashy info.

### Basic Information

- Version with issue: 4.5.0.617
- IDE: Visual Studio for Mac 8.4.8 (build 2)
- Platform Target Frameworks: 
  - iOS:  13.3
  - Android: 9.0

**Comment by jsuarezruiz** [comment]
Tested. The results are different between Android and iOS:

Android:

- Navigating: https://www.microsoft.com/
- Navigating: https://www.microsoft.com/es-es/
- Navigated: https://www.microsoft.com/es-es/

iOS:
- Navigating: https://www.microsoft.com/
- Navigating: https://www.microsoft.com/es-es/
- Navigating: about:blank
- Navigating: https://login.live.com/me.srf?wa=wsignin1.0&wreply=https%3A%2F%2Fwww.microsoft.com&uaid=...
- Navigating: https://www.microsoft.com/store/buy/cartcount
- Navigated: https://www.microsoft.com/es-es/

**Comment by hartez** [comment]
Not sure this is a bug - the platforms are using their respective native web views, which means that they're probably sending different user agents. So the server is treating them differently based on their capabilities.

**Comment by Trivias91** [comment]
I'm also seeing an inconsistency in that POST requests are triggering Navigating events in iOS, but on Android those same POST requests do not trigger Navigating events. (Forms 4.7.0.1239, Android 10,  Not tested on other versions of either)

**Comment by kensykora** [comment]
We are seeing a similar issue -- not quite the same steps to reproduce, but interesting behavior. It is making it challenging for us to support a web view loading indicator.

Our use case is to support a "loading" spinner on the webview where we show/hide during navigating/navigated events respectively

**On Android Steps to Reproduce**:

1. Set a web view's Source property
2. Observe Navigating event fires
3. While navigation is still occurring (slow loading website), set the web view's source property again

**Expected**: one of these two options:
(1)  Navigating  (first page did not complete, so no Navigated event)
      Navigated (for completion)

OR  

(2)  Navigated (for first source, Result = Cancelled or similar)
       Navigating (for new source)
       Navigated (for completion)

**Actual**: 

Navigating (for new source)
Navigated: Result = Success (for old source)
Navigated: Result = Success (for new source)

---

Based on the order of these events in the actual, there's no way I can determine that the page is still in a "loading" state.

**Comment by kensykora** [comment]
We are seeing a similar scenario in iOS when navigating to a link that has `<a href="#">` which points at itself

iOS: 
 - Navigating (no corresponding Navigated event)

Android:
 - Navigating
 - Navigated

**Comment by kensykora** [comment]
> Not sure this is a bug - the platforms are using their respective native web views, which means that they're probably sending different user agents. So the server is treating them differently based on their capabilities.

@hartez I don't think this has anything to do with the servers, but rather then underlying native implementations of the web viewers in both iOS and Android. Xamarin exposes an abstraction layer for Navigating/Navigated events which iOS WkWebView and Android WebView implement in their respective renderers. It's my theory that there is an underlying issue with the renderers that are handling these events inconsistently. If we are aiming for cross platform compatibility, my Xamarin Forms code needs to rely on these events firing similarly and with some consistency. It doesn't need to be exact, but I have outlined above a few scenarios where the inconsistency is causing issues in our Xamarin Forms app.

Again, our use case is for showing and hiding a loading indicator while page navigation is occurring in the webview, since there is no Loading Indicator option with the webviews themselves. If we incorrectly receive a Navigated event ([case 1](https://github.com/xamarin/Xamarin.Forms/issues/10350#issuecomment-702898011)) or never receive it at all ([case 2](https://github.com/xamarin/Xamarin.Forms/issues/10350#issuecomment-704398499)), it leaves me with a buggy implementation of this feature.

I think my workaround at this point is to either override the rendering or implement my own renderer.

**Comment by captainjono** [comment]
@kensykora 
https://github.com/xamarin/Xamarin.Forms/issues/14373#issuecomment-870153754

here is a loadUrl in the WkWebView renderer for cookie purposes which i think could lead to different behaviour
